frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2868,10 +2868,40 @@ static bool resolveFnCallSpecial(Context* context,
                                  const AstNode* astForErr,
                                  const CallInfo& ci,
                                  QualifiedType& exprTypeOut) {
-  // TODO: cast
   // TODO: .borrow()
   // TODO: chpl__coerceCopy
 
+  // explicit param casts are resolved here
+  if (ci.isOpCall() && ci.name() == USTR(":")) {
+    auto src = ci.actual(0).type();
+    auto dst = ci.actual(1).type();
+
+    bool isDstType = dst.kind() == QualifiedType::TYPE;
+    bool isParamTypeCast = src.kind() == QualifiedType::PARAM && isDstType;
+
+    if (isParamTypeCast) {
+        exprTypeOut = Param::fold(context, uast::PrimitiveTag::PRIM_CAST,
+                                  src, dst);
```

Comment:
There are some param casts that aren't legal, right? I think? For example, `myEnum.red: nothing` ? I suspect that most are legal but we'll need some checking that it's a valid cast somewhere & I don't think we have that yet. I'd expect it'd be better to check it here than in `Param::fold` but either way could work.

Reply:
I added the rules to catch illegal casts involving enum to nothing, abstract enum to int, and int to abstract enum. I found those in prod from `functionResolution.cpp`, although there may be others defined in other places